app/models/solid_errors/occurrence.rb

```diff
@@ -23,5 +23,16 @@ def parse_backtrace(backtrace)
     def send_email
       ErrorMailer.error_occurred(self).deliver_later
     end
+
+    def should_send_email?
+      SolidErrors.send_emails? && SolidErrors.email_to.present? && under_limit?
+    end
+
+    def under_limit?
+      return true if error.occurrences.count == 1
+      return true if !SolidErrors.one_email_per_occurrence
+
+      error.occurrences.where(created_at: error.prev_resolved_at..).one?
+    end
```

Comment:
I actually think we can just collapse this into one method:
```suggestion
    def should_send_email?
      return false unless SolidErrors.send_emails?
      return false unless SolidErrors.email_to.present?
      return true if error.occurrences.count == 1
      return true if not SolidErrors.one_email_per_occurrence
      
      error.occurrences.where(created_at: error.prev_resolved_at..).one?
    end
```

Reply:
Totally! Just committed a change for this. I went with `return true unless SolidErrors.one_email_per_occurrence?` to both match the other `SolidErrors.` checks and I made the method a predicate like `send_emails?` as that felt more consistent.